src/login/logind.c

```diff
@@ -410,74 +411,86 @@ static int parse_fdname(const char *fdname, char **session_id, dev_t *dev) {
         return 0;
 }
 
-static int manager_attach_fds(Manager *m) {
-        _cleanup_strv_free_ char **fdnames = NULL;
-        int n, i, fd;
+/* Release stored fd from PID 1.  Note DRM fds can hold a lot of memory.
+ * Let's at least implement this without requiring a heap allocation :). */
 
-        /* Upon restart, PID1 will send us back all fds of session devices
-         * that we previously opened. Each file descriptor is associated
-         * with a given session. The session ids are passed through FDNAMES. */
+static void fdstore_remove(const char *fdname) {
+        char message[sizeof("FDSTOREREMOVE=1\n" "FDNAME=\n") + 255];
```

Comment:
i don't like arbitrary sized buffers I must say, i.e. the "255" here...

I think strjoina() would work here much nicer, no?